mfs.go

```diff
@@ -0,0 +1,65 @@
+package iface
+
+import (
+	"context"
+	"io"
+	"os"
+)
+
+// MfsPath
+//
+// Note that Path and ResolvedPath interfaces satisfy this interface
+//
+// This interface is inspired by https://godoc.org/github.com/src-d/go-billy,
+// but doesn't implement it
+type MfsPath interface {
+	// String returns the path as a string.
+	String() string
+
+	// Mutable returns false if the data pointed to by this path in guaranteed
+	// to not change.
+	//
+	// Note that resolved mutable path can be immutable.
+	Mutable() bool
+}
+
+// TODO: ParseMfsPath
+
+type MfsAPI interface {
```

Comment:
Should we name this `Mfs` or would `Files` / something else be a better name?

Reply:
I think `Files` is more appropriate if we're specifically targeting the same abstract root as `ipfs files` on the node.
If this API is not intending to handle arbitrary mfs roots, I think the name can cause some confusion with the existing `mfs` pkg.

Reply:
There really isn't anything stopping this from handling arbitrary roots, other than lack of support via http api.

We could keep the `Mfs*` names on interfaces and expose it as `Files() MfsAPI` on the CoreAPI interface (and then explain it in a nice docstring)